File: 1_general_coding_logic/question_2/leap_years.php
<!DOCTYPE html>
<html lang="en">

<body>

<?php

/**
* Returns TRUE if year is a leap year. Otherwise, FALSE.
*/
function leap_year(int $year) {
	if (($year % 4) != 0) {
    	return FALSE;
    } elseif (($year % 100) != 0) {
    	return TRUE;
    } elseif (($year % 400) != 0) {
		return FALSE;
	} else {
		return FALSE;
	}
}
	
$current_year = 2021;
$count = 0;

while ($count != 20) {
	if (leap_year($current_year)) {
		echo $current_year;
		echo "<br>";
		$count += 1;
	}
	$current_year += 1;
}

?>

</body>
</html>

File: 1_general_coding_logic/question_2/leap_years.py
def leap_year(year):
    """Returns True if year is a leap year.
    Otherwise, False."""

    if (year % 4) != 0:
        return False
    elif (year % 100) != 0:
        return True
    elif (year % 400) != 0:
        return False
    else:
        return False


def next_leap_years(year, n):
    """Prints next n leap years starting at given year."""

    count = 0

    while count != n:
        if leap_year(year):
            print(year)
            count += 1
        year += 1


next_leap_years(2021, 20)  # next 20 leap years
